File: algo/find-the-duplicate-number/findDuplicate.c
int findDuplicate(int* nums, int numsSize) {
    int slow, fast, meet;
    
    slow = nums[0];
    fast = nums[nums[0]];
    
    while (slow != fast) {
        slow = nums[slow];
        fast = nums[nums[fast]];
    }
    
    meet = 0;
    while (meet != slow) {
        slow = nums[slow];
        meet = nums[meet];
    }
    
    return meet;
}

File: algo/find-the-duplicate-number/findDuplicate.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        slow, fast, meet = nums[0], nums[nums[0]], 0
        
        while slow != fast:
            slow, fast = nums[slow], nums[nums[fast]]
        
        while meet != slow:
            slow, meet = nums[slow], nums[meet]
            
        return meet
